mozilla-ai/lumigator
# Resolving pathing and build dependencies between lm-buddy and Lumigator

## What's changing

Now that we've imported lm-buddy into Lumigator, we need to make sure it builds correctly both locally on Mac and on remote servers with Linux distributions. This PR addresses two issues: 

1) The relative pathing is incrroect for lm buddy to be imported into Lumigator, this change is from `lumigator.python.mzai.lm_buddy to `mzai.lm_buddy` across all lm-buddy artifacts. 

2) The second issue is that now that we have two sets of dependencies, it's not clear how they should be resolved in light of Pants and there are import resolution issues because now both packages use, for example, Ray. 

## How to test it

To reproduce this problem:

A. Fetch branch
B. run `pants test ::` at the root of the repo, you should see issues resolving both ray and PyTorch (although this might be unrelated)

## Related Jira Ticket

## Additional notes for reviewers

In order to get these tests to pass, build a new container and test on Coreweave and merge this PR, we need to decide: 

1. How to do dependency resolution between packages in the short-term
2. Whether we should keep PyTorch as a top-level platform dependency or relegate to the list of dependencies shipped to Ray
3. Whether to keep Pants  <--- this can perhaps be folded into a next PR


## I already...


cc @javiermtorres @ividal @aittalam
